src/Model/CallbackIterator.php

```diff
@@ -49,7 +49,7 @@ public function __construct(Traversable $traversable, Closure $callback)
     #[ReturnTypeWillChange]
     public function current()
     {
-        return ($this->callback)($this->iterator->current());
+        return ($this->callback)($this->iterator->current(), $this->iterator->key());
```

Comment:
I noticed that our `CallbackIterator` only passes the value to the callback, but not the key. This change brings the iterator in line with `CallbackFilterIterator` which also passes both. Note that the latter also passes the whole iterator object as well, which I can add for the sake of feature parity, but it isn't required for this PR to work properly.